server/interpreter.js

```diff
@@ -2048,6 +2047,19 @@ Interpreter.prototype.executeException = function(error) {
   throw realError;
 };
 
+/**
+ * Create and push a new state onto the statestack.
+ * @param {!Object} node AST node for the state.
+ * @return {!Object} New state.
+ */
+Interpreter.prototype.pushNode_ = function(node) {
```

Comment:
`stateStack` is a plain array.  To put it there means one of:

* Create the function elsewhere, and copy it onto the array each time it is created.
* Attempt to subclass Array.
* Create a wrapper object that contains the function, and an array, and an initializer.

Each of these seems to increase complexity.  Not sure what you have in mind.